intelmq/bots/parsers/abusech/parser_ip.py

```diff
@@ -6,36 +6,97 @@
  - https://feodotracker.abuse.ch/blocklist/
  - https://zeustracker.abuse.ch/blocklist.php
 """
-
+import re
 
 import dateutil
 
 from intelmq.lib.bot import ParserBot
+from intelmq.lib import utils
+from intelmq.lib.exceptions import PipelineError
 
-SOURCE_FEEDS = {'https://feodotracker.abuse.ch/blocklist/?download=ipblocklist': 'Cridex',
-                'https://zeustracker.abuse.ch/blocklist.php?download=ipblocklist': 'Zeus',
-                'https://zeustracker.abuse.ch/blocklist.php?download=badips': 'Zeus'}
+FEEDS = {
+    'https://feodotracker.abuse.ch/downloads/ipblocklist.csv': {
+        'format': [
+            'time.source',
+            'source.ip',
+            'source.port',
+            'malware.name'
+        ],
+        'malware': 'Cridex'
+    },
+    'https://zeustracker.abuse.ch/blocklist.php?download=ipblocklist': {
+        'format': [
+            'source.ip'
+        ],
+        'malware': 'Zeus'
+    },
+    'https://zeustracker.abuse.ch/blocklist.php?download=badips': {
+        'format': [
+            'source.ip'
+        ],
+        'malware': 'Zeus'
+    }
+}
 
 
 class AbusechIPParserBot(ParserBot):
     lastgenerated = None
 
+    def parse(self, report: dict):
+        lines = utils.base64_decode(report.get("raw")).splitlines()
+        regex = re.compile(r'^#+.*')  # filter out comments
```

Comment:
Can be moved one level up to the class (doesn't need to be re-compiled on every run)

Reply:
Thank you, I will change that.